### ISSUE TITLE
[469b] Add in-built send UZ files

### ISSUE THREAD

**Issue by SeriousBuggie** [OldUnreal/UnrealTournamentPatches#934]
Current approach for send UZ files is very fragile:
1. You need HTTP server (when modern world try goes rid of it and make all HTTPS)
2. You need not forget update UZ files
3. You need proper setup in INI files.

I suggest make some folder with UZ cache files.
When map start server compress all files, which not here, or older from original.
It must be pretty fast, and allow compress only new files.
During gameplay, server provide old-compatible UZ redirect to HTTP server, where HTTP server is gameserver itself.
On it is send UZ files uncapped, as direct send work.

This allow support old clients, and transparent enable UZ compression for all servers, which greatly increase QoL admins and users, since now many of them suffer from many steps, need for setup it or by luck support from server side and long wait download.

**Comment by an-eternity** [comment]
Is it possible to implement direct work with UZ-compressed files runtime? Decompression process is quite fast, so it shouldn't be a problem, and would have many advantages, such as much less disk size would be taken on the server and synchronization tasks (which is a pain for many beginner admins out there) to keep UZ archive up to date would no longer be needed.

Also, can this functionality be implemented through native mutator, by overriding DLO function, or somehow else?..

**Comment by SeriousBuggie** [comment]
Not get questions. As I suggest there already be folder for UZ cache which can be auto managed. Possible with involve some INI file where stored exact size and GUID of stored UZ file. So can easy detect mismatch and make new UZ version on map start.

**Comment by an-eternity** [comment]
My question - is it possible to get rid of uncompressed files at all, and to load packages from compressed files directly, decompressing them runtime.

**Comment by SeriousBuggie** [comment]
No. It is huge impact to load size. Decode is not fast enough and CPU intensive, which is not allowed for 20+ old years game, which must run on old hardware.

**Comment by CacoFFF** [comment]
Due to possible network incompatibilities, this is only feasible if we change version number to 470.

XC_Engine already has this feature (as well as LZMA file support) but it's implemented in a very hacky way.

**Comment by SeriousBuggie** [comment]
I not see how it be incompatible with anything.

Lets look to example:
Current approach (barbie server)
```
DevNet: PendingLevel received: DLMGR CLASS=IpDrv.HTTPDownload PARAMS=http://barbies.world:46139/ COMPRESSION=1
DevNet: PendingLevel received: DLMGR CLASS=Engine.ChannelDownload PARAMS=Enabled COMPRESSION=0
```
What we see - `http://barbies.world:46139/` as UZ for external http server.

What be instead (suggestion):
```
DevNet: PendingLevel received: DLMGR CLASS=IpDrv.HTTPDownload PARAMS=http://barbies.world:12345/ COMPRESSION=1
DevNet: PendingLevel received: DLMGR CLASS=Engine.ChannelDownload PARAMS=Enabled COMPRESSION=0
```
port 12345 (can be any) listen internal webserver from UT99 server. Same as for WebAdmin, but server here .uz files, from some uz folder. 

On this uz folder compressed actual list of files, which based on:
`DevNet: PendingLevel received: USES GUID=D2AFD44011D49539500053B3B8735104 PKG=Gta2 FLAGS=1 SIZE=1826555 GEN=1 FNAME=Gta2.u`
OFC update only if uz file outdated. For this can store in ini file near (like in cache) additional info for each uz file, if this need,

**Comment by CacoFFF** [comment]
Thought of running a HTTP server internally in the game, realized it was a spectacularly awful idea for multiple reasons.

The very simple solution is to run this over the Engine.ChannelDownload and move the file decompression functions from IpDrv to Engine, additionally a client should be able to request files by sending their preferred compression algorithms in the request.

XC_Engine already runs an asynchronous subsystem where packages currently relevant to the server are queued for compression with an optional on-disk cache, it's LZMA only but it can be ported to the main game to support UZ and more.

Extremely simple stuff:
![imagen](https://user-images.githubusercontent.com/15526924/179439322-07b40a35-5e0d-46d2-92a3-97b703a6bd3e.png)
